File: src/tools/tui.ts
import { QcError } from "../errors.js";
import type { ToolName } from "./types.js";

/** Interactive resume only: no JSON, force-allow, model, thinking, or workspace flags. */
export function buildPiTuiArgs(nativeId: string): string[] {
  return ["--session-id", nativeId];
}

export function buildCursorTuiArgs(nativeId: string): string[] {
  return ["--resume", nativeId];
}

export function buildClaudeTuiArgs(nativeId: string): string[] {
  return ["--resume", nativeId];
}

export function buildOpenCodeTuiArgs(nativeId: string): string[] {
  return ["-s", nativeId];
}

export function buildAntigravityTuiArgs(nativeId: string): string[] {
  return ["--conversation", nativeId];
}

export function buildTuiArgs(tool: ToolName, nativeId: string): string[] {
  switch (tool) {
    case "pi":
      return buildPiTuiArgs(nativeId);
    case "cursor":
      return buildCursorTuiArgs(nativeId);
    case "claude":
      return buildClaudeTuiArgs(nativeId);
    case "opencode":
      return buildOpenCodeTuiArgs(nativeId);
    case "antigravity":
      return buildAntigravityTuiArgs(nativeId);
    default: {
      const _exhaustive: never = tool;
      throw new QcError(`unknown tool '${String(_exhaustive)}'`);
    }
  }
}

File: src/tools/tui.rs
use crate::errors::QcError;
use crate::tools::types::ToolName;

/// Interactive resume only: no JSON, force-allow, model, thinking, or workspace flags.
pub fn build_pi_tui_args(native_id: &str) -> Vec<String> {
    vec!["--session-id".into(), native_id.to_string()]
}

pub fn build_cursor_tui_args(native_id: &str) -> Vec<String> {
    vec!["--resume".into(), native_id.to_string()]
}

pub fn build_claude_tui_args(native_id: &str) -> Vec<String> {
    vec!["--resume".into(), native_id.to_string()]
}

pub fn build_opencode_tui_args(native_id: &str) -> Vec<String> {
    vec!["-s".into(), native_id.to_string()]
}

pub fn build_antigravity_tui_args(native_id: &str) -> Vec<String> {
    vec!["--conversation".into(), native_id.to_string()]
}

pub fn build_codex_tui_args(native_id: &str) -> Vec<String> {
    vec!["resume".into(), native_id.to_string()]
}

pub fn build_tui_args(tool: ToolName, native_id: &str) -> Result<Vec<String>, QcError> {
    Ok(match tool {
        ToolName::Pi => build_pi_tui_args(native_id),
        ToolName::Cursor => build_cursor_tui_args(native_id),
        ToolName::Claude => build_claude_tui_args(native_id),
        ToolName::Opencode => build_opencode_tui_args(native_id),
        ToolName::Antigravity => build_antigravity_tui_args(native_id),
        ToolName::Codex => build_codex_tui_args(native_id),
    })
}
